deployment/datastore/abstract.go

```diff
@@ -0,0 +1,77 @@
+package datastore
```

Comment:
I think we need to come up with a better name than datastore. I feel like this term in too generic and can lead to more confusion from the CLD users

Reply:
This was also addressed in a separate conversation. While `datastore` is admittedly quite generic, it currently seems to be the only name that’s "good enough" to describe the data structure used to store address-related data, contract metadata, and other domain-specific information.

Other names that were considered are Repository. We can always revisit this decision later if a more suitable name comes up.